go-controller/pkg/ovn/base_network_controller_namespace.go

```diff
@@ -386,6 +387,14 @@ func (bsnc *BaseNetworkController) removeRemoteZonePodFromNamespaceAddressSet(po
 	podDesc := fmt.Sprintf("pod %s/%s/%s", bsnc.GetNetworkName(), pod.Namespace, pod.Name)
 	podIfAddrs, err := util.GetPodCIDRsWithFullMask(pod, bsnc.NetInfo)
 	if err != nil {
+		// maybe the pod is not scheduled yet or addLSP has not happened yet, so it doesn't have IPs.
+		// let us ignore deletion failures for podIPs not found because
+		// there is nothing more we can do here.
+		if errors.Is(err, util.ErrNoPodIPFound) {
+			klog.Errorf("Unable to remove remote zone pod's %s/%s IP address from the "+
+				"namespace address-set, err: %v", pod.Namespace, pod.Name, err)
+			return nil
```

Comment:
At most I would use a warning here, not an error. It wouldn't even be a warning for secondary network with no IPAM. And there are already a lot of warnings being logged from within `GetPodCIDRsWithFullMask`.

So ideally I would consolidate all of those to a single warning if the network has IPAM.

But @maiqueb was already working on that here: https://github.com/ovn-org/ovn-kubernetes/pull/3612

Reply:
ack will change this to warning..